File: OTCDerivativesCalculatorModule/Project_CSharp/XmlBuilder/XmlSerializedClass/fpml-msg-5-4/EventStatusItem.cs
using System;
using System.Text;
using System.Collections.Generic;
using System.Xml;

namespace FpmlSerializedCSharp
{
    public class EventStatusItem : ISerialized
    {
        public EventStatusItem(XmlNode xmlNode)
        {
            XmlNode eventIdentifierNode = xmlNode.SelectSingleNode("eventIdentifier");
            
            if (eventIdentifierNode != null)
            {
                if (eventIdentifierNode.Attributes["href"] != null || eventIdentifierNode.Attributes["id"] != null) 
                {
                    if (eventIdentifierNode.Attributes["id"] != null) 
                    {
                        eventIdentifierIDRef_ = eventIdentifierNode.Attributes["id"].Value;
                        EventIdentifier ob = new EventIdentifier(eventIdentifierNode);
                        IDManager.SetID(eventIdentifierIDRef_, ob);
                    }
                    else if (eventIdentifierNode.Attributes["href"] != null)
                    {
                        eventIdentifierIDRef_ = eventIdentifierNode.Attributes["href"].Value;
                    }
                    else
                    {
                        eventIdentifier_ = new EventIdentifier(eventIdentifierNode);
                    }
                }
                else
                {
                    eventIdentifier_ = new EventIdentifier(eventIdentifierNode);
                }
            }
            
        
            XmlNode statusNode = xmlNode.SelectSingleNode("status");
            
            if (statusNode != null)
            {
                if (statusNode.Attributes["href"] != null || statusNode.Attributes["id"] != null) 
                {
                    if (statusNode.Attributes["id"] != null) 
                    {
                        statusIDRef_ = statusNode.Attributes["id"].Value;
                        EventStatus ob = new EventStatus(statusNode);
                        IDManager.SetID(statusIDRef_, ob);
                    }
                    else if (statusNode.Attributes["href"] != null)
                    {
                        statusIDRef_ = statusNode.Attributes["href"].Value;
                    }
                    else
                    {
                        status_ = new EventStatus(statusNode);
                    }
                }
                else
                {
                    status_ = new EventStatus(statusNode);
                }
            }
            
        
        }
        
    
        #region EventIdentifier_
        private EventIdentifier eventIdentifier_;
        public EventIdentifier EventIdentifier_
        {
            get
            {
                if (this.eventIdentifier_ != null)
                {
                    return this.eventIdentifier_; 
                }
                else if (this.eventIdentifierIDRef_ != null)
                {
                    eventIdentifier_ = IDManager.getID(eventIdentifierIDRef_) as EventIdentifier;
                    return this.eventIdentifier_; 
                }
                else
                {
                      return this.eventIdentifier_; 
                }
            }
            set
            {
                if (this.eventIdentifier_ != value)
                {
                    this.eventIdentifier_ = value;
                }
            }
        }
        #endregion
        
        public string eventIdentifierIDRef_ { get; set; }
        #region Status_
        private EventStatus status_;
        public EventStatus Status_
        {
            get
            {
                if (this.status_ != null)
                {
                    return this.status_; 
                }
                else if (this.statusIDRef_ != null)
                {
                    status_ = IDManager.getID(statusIDRef_) as EventStatus;
                    return this.status_; 
                }
                else
                {
                      return this.status_; 
                }
            }
            set
            {
                if (this.status_ != value)
                {
                    this.status_ = value;
                }
            }
        }
        #endregion
        
        public string statusIDRef_ { get; set; }
        
    
        
    
    }
    
}

File: OTCDerivativesCalculatorModule/Project_Cpp/lib_static/FpmlSerialized/GenClass/fpml-msg-5-4/EventStatusItem.cpp
// EventStatusItem.cpp 
#include "EventStatusItem.hpp"
#ifdef ConsolePrint
    #include <iostream>
#endif 

namespace FpmlSerialized {

EventStatusItem::EventStatusItem(TiXmlNode* xmlNode)
: ISerialized(xmlNode)
{
    #ifdef ConsolePrint
        std::string initialtap_ = FileManager::instance().tap_;
        FileManager::instance().tap_.append("   ");
    #endif 
   //eventIdentifierNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* eventIdentifierNode = xmlNode->FirstChildElement("eventIdentifier");

   if(eventIdentifierNode){eventIdentifierIsNull_ = false;}
   else{eventIdentifierIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- eventIdentifierNode , address : " << eventIdentifierNode << std::endl;
   #endif
   if(eventIdentifierNode)
   {
      if(eventIdentifierNode->Attribute("href") || eventIdentifierNode->Attribute("id"))
      {
          if(eventIdentifierNode->Attribute("id"))
          {
             eventIdentifierIDRef_ = eventIdentifierNode->Attribute("id");
             eventIdentifier_ = boost::shared_ptr<EventIdentifier>(new EventIdentifier(eventIdentifierNode));
             eventIdentifier_->setID(eventIdentifierIDRef_);
             IDManager::instance().setID(eventIdentifierIDRef_,eventIdentifier_);
          }
          else if(eventIdentifierNode->Attribute("href")) { eventIdentifierIDRef_ = eventIdentifierNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { eventIdentifier_ = boost::shared_ptr<EventIdentifier>(new EventIdentifier(eventIdentifierNode));}
   }

   //statusNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* statusNode = xmlNode->FirstChildElement("status");

   if(statusNode){statusIsNull_ = false;}
   else{statusIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- statusNode , address : " << statusNode << std::endl;
   #endif
   if(statusNode)
   {
      if(statusNode->Attribute("href") || statusNode->Attribute("id"))
      {
          if(statusNode->Attribute("id"))
          {
             statusIDRef_ = statusNode->Attribute("id");
             status_ = boost::shared_ptr<EventStatus>(new EventStatus(statusNode));
             status_->setID(statusIDRef_);
             IDManager::instance().setID(statusIDRef_,status_);
          }
          else if(statusNode->Attribute("href")) { statusIDRef_ = statusNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { status_ = boost::shared_ptr<EventStatus>(new EventStatus(statusNode));}
   }

    #ifdef ConsolePrint
        FileManager::instance().tap_ = initialtap_;
    #endif 
}
boost::shared_ptr<EventIdentifier> EventStatusItem::getEventIdentifier()
{
   if(!this->eventIdentifierIsNull_){
        if(eventIdentifierIDRef_ != ""){
             return boost::shared_static_cast<EventIdentifier>(IDManager::instance().getID(eventIdentifierIDRef_));
        }else{
             return this->eventIdentifier_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<EventIdentifier>();
   }
}
boost::shared_ptr<EventStatus> EventStatusItem::getStatus()
{
   if(!this->statusIsNull_){
        if(statusIDRef_ != ""){
             return boost::shared_static_cast<EventStatus>(IDManager::instance().getID(statusIDRef_));
        }else{
             return this->status_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<EventStatus>();
   }
}
}

